README.md

```diff
@@ -12,3 +12,14 @@ This framework enables AWS IoT cloud connectivity with ESP32 based platforms usi
   - ESP-IDF can be downloaded from https://github.com/espressif/esp-idf/
   - ESP-IDF v3.1 and above is recommended version
 - Please refer to [example README](examples/README.md) for more information on setting up examples
+
+
+## Using this library with an ATECC608a
+This port contains support for using a secure element chip from microchip.  This will store the private key used for TLS communication in the secure element.  To use this function, perform the following tasks:
+
+1. add [esp-cryptoauthlib](https://github.com/espressif/esp-cryptoauthlib) to your project's components, and make sure it is configured
+1. Turn on AWS_IOT_USE_HARDWARE_SECURE_ELEMENT in the sdkconfig of your project.
+1. Ensure that you call `atcab_init` to initialise the microchip library before you use the AWS IoT library
+1. When connecting to AWS IoT Core, change your connection configuration to;
+  1. If you wish to use the certificate stored in the ATECC608a, set `mqttInitParams.pDeviceCertLocation = "#"`.  You may still use a certificate stored in code or on the filesystem, if you wish
+  1. Tell the IoT library to find the Private key in a slot by setting `mqttInitParams.pDevicePrivateKeyLocation = “#0”`, where the digit 0 indicates the slot in the ATECC608a that should be used to sign requests.  In most circumstances, the key will be stored in slot 0
```

Comment:
```suggestion

## Using this library with an ATECC608A
This port contains support for using a `secure element` chip from microchip i.e. `ATECC608A`. This will store the private key used for TLS communication in the `secure element`. The module [ESP32-WROOM-32SE](https://www.espressif.com/sites/default/files/documentation/esp32-wroom-32se_datasheet_en.pdf) which integrates the `ATECC608A` can be readily used for this purpose. To use the `secure element` for the TLS, configure the project with help of the following steps.
> To use the `secure element`(ATECC608A) for the TLS connection, it must be already configured. Please refer to the [eps_cryptoauth_utility](https://github.com/espressif/esp-cryptoauthlib/tree/master/esp_cryptoauth_utility) for more details.
### Project Configuration

1. Add [esp-cryptoauthlib](https://github.com/espressif/esp-cryptoauthlib) as a components to your project with the help of following commands,
    (First change directory (cd) to your project directory)
```
    mkdir components
    cd components
    git clone https://github.com/espressif/esp-cryptoauthlib.git
```
2. Enable support for hardware secure element with the following menuconfig option,

    `menuconfig->Component config->Amazon Web Services IoT Platform->Allow hardware secure element for certificates`
3. Ensure that you call the function `atcab_init` in the application to initialise the microchip library before you use the AWS IoT library.
4. Project TLS configuration

    i) To use the client certificate stored in the ATECC608A, set `mqttInitParams.pDeviceCertLocation = "#"`.This will only work for `Trust&GO` type of ATECC608A chip. For other types of ATECC608A chips, you will need to provide the certificate externally(embed the certificate).

    ii) Tell the IoT library to find the client private key in a slot by setting `mqttInitParams.pDevicePrivateKeyLocation = “#0”`, where the digit 0 indicates the slot in the ATECC608A in which the private key is stored.  In most circumstances, the key will be stored in slot 0.
```

Reply:
I have added few points and edited the README, please feel free to add or change any stuff which you think should not be here.